### DISCUSSION TITLE
Is the FLARE community interested in a Minecraft-style-graphics ARPG?

### DISCUSSION THREAD

**Discussion by Des-Nerger** [flareteam/flare-game#953] [Ideas]
Are you people interested in a such cuboid ARPG?
I'm asking because I think that kind of assets would be easier / cheaper for the community to produce and fit into each other.
Which means people could generate more content for FLARE.
For instance, the community of libre implementations of Minecraft (e.g. "MineClone2" running on top of the Minetest engine) has these models: https://github.com/22i/minecraft-voxel-blender-models, among the other assets (like the textures for the blocks). I imagine those would be enough to set up a basic Action-RPG which in theory could become more lively developed than the Empyrean Campaign currently is. Besides, the cuboids are also easier to work with when it comes to the rogue/moria-style random dungeon generators which was a staple of Diablo I 🤔...

And talking about the references, do you guys think the `Minecraft Dungeons` game was a failure, or did it really have a potential? Also, there's the `Riverbond` game to consider of...

**Comment by Des-Nerger** [comment]
My hypothesis is that the whole appeal of pre-rendered isometric RPG engines like flare-engine or GemRB comes from the ability for a player to not see / bear-with the polygonal nature of the models (while also not paying with extra GPU / electricity resources for their triangles' rasterization real-time). Cuboid models however, like those from MineClone2, _do_ have the unconcealed polygonness, therefore I think you won't ever find them appealing in the context of FLARE. But that all is just my wild guess; may be wrong.

**Comment by shinobody** [comment]
I think the question to consider is if it would be a good game. And I mean, Minecraft Dungeons is currently one of more popular ARPGs out there, currently easily competing with games like Diablo. I see no reason one couldn't make "Minetest Dungeons' or something like that.

Also I'm not sure why people would find polygons unappealing - right now games with 'low poly' aesthetic are topping indie game sales charts.
This was maybe true in like late 90s when prerendered aesthetic was new, but by now pre-rendered art in Flare is as much of a retro throwback as low-poly art in games like Ultrakill or Corn Kidz 64 is.

**Reply by Des-Nerger** [reply]
> not sure why people would find polygons unappealing

Not unappealing in general, but unappealing for a flare-engine game. I meant, with clear low-poly, using the flare-engine (at least to me) doesn't feel much advantageous compared to a realtime-3D engine. There'd be only drawbacks: with a pre-rendered engine one had to store all those "unnecessary" sprites, projected tiles, etc. Just "a good game" could be made somewhere else; there must be a good justification for why flare-engine and its tooling ecosystem were chosen, exactly. Maybe it could be "pre-calculated light / shadows", but I doubt so.

**Reply by Des-Nerger** [reply]
> Maybe it could be "pre-calculated light / shadows", but I doubt so.

Oh, I guess I missed an important point.

@dorkster, @Danimal696, am I right that the most recent renderings of the flare-game 3D assets were done using the "cycles" renderer of Blender?

If so, given that it's a path tracing renderer, the quality of its images will just always be a head above of what any real-time rasterizer can ever give, while, at the same time, not requiring the heavy computation to be performed on the end-user side. So the low-poly / cuboid feel seems to be a separate issue.
I wonder if cuboid / voxel world and creatures could have a close enough (to flare-game's) lighting aesthetics when rendered through the Cycles renderer :thinking:... ![image](https://github.com/flareteam/flare-game/assets/7359990/3b9b5795-9ee4-46a1-a94e-42d0a616e146) [[Dungeon from Minecraft, rendered in Blender Cycles](https://www.reddit.com/r/Minecraft/comments/jlh6jx/dungeon_from_minecraft_rendered_in_blender_cycles/)]

p.s. Oh, wait, the first versions of FLARE were done in Blender v2.49, when the Cycles renderer just didn't exist yet! --> I've been confusing myself here. But still, its renderer was a ray-tracer, and not a rasterizer, wasn't it?.. :thinking:

**Comment by Des-Nerger** [comment]
Finally started trying making something.

https://github.com/user-attachments/assets/8c46b45e-373f-4bf8-be1a-b2f1ab3d2ce7
